stl/inc/flat_set

```diff
@@ -263,7 +263,15 @@ public:
     template <_Container_compatible_range<_Kty> _Rng>
     void insert_range(_Rng&& _Range) {
         const size_type _Old_size = size();
-        _Get_cont().append_range(_STD forward<_Rng>(_Range));
+
+        _Container& _Cont = _Get_cont();
+        if constexpr (requires { _Cont.append_range(_STD forward<_Rng>(_Range)); }) {
+            _Cont.append_range(_STD forward<_Rng>(_Range));

```

Comment:
`insert_range(end,...)` is unconditionally required for sequence containser. however, I think we don't have to check for that chance, asthis is just an optimization, and it is likely that a container that doesn't support `append_range` cannot support `insert_range` either.

Reply:
Also, I find these funtions lack `forward` (required by the standard):
https://github.com/microsoft/STL/blob/48eedd369d6c88130fff4d15afb88f016fa6e648/stl/inc/stack#L115-L117
https://github.com/microsoft/STL/blob/48eedd369d6c88130fff4d15afb88f016fa6e648/stl/inc/queue#L125-L127

Reply:
Should we forward here? It seems that `std::forward` merely rejects non-borrowed-range rvalues, which defends dangling.

But we are eagerly consuming ranges here, so it seems to me that we can just use the ranges as lvalues for containers. CC-ing @cor3ntin @ericniebler @CaseyCarter.